MatrixSDK/Background/Store/MXSyncResponseStoreManager.swift

```diff
@@ -0,0 +1,292 @@
+                        leftRoomSync.timeline?.limited = true
+                    }
+                }
+                
+                //  handle old limited timelines
+                cachedSyncResponse.syncResponse.rooms.join.filter({ $1.timeline?.limited == true }).forEach { (roomId, _) in
+                    if let joinedRoomSync = newSyncResponse.rooms.join[roomId] {
+                        //  mark new timeline as limited too, to avoid losing value of limited
+                        joinedRoomSync.timeline?.limited = true
+                    }
+                }
+                cachedSyncResponse.syncResponse.rooms.leave.filter({ $1.timeline?.limited == true }).forEach { (roomId, _) in
+                    if let leftRoomSync = newSyncResponse.rooms.leave[roomId] {
+                        //  mark new timeline as limited too, to avoid losing value of limited
+                        leftRoomSync.timeline?.limited = true
+                    }
+                }
+                
+                // Merge the new sync response to the old one
+                var dictionary = NSDictionary(dictionary: cachedSyncResponse.syncResponse.jsonDictionary())
+                dictionary = dictionary + NSDictionary(dictionary: newSyncResponse.jsonDictionary())
+                
+                // And update it to the store.
+                // Note we we care only about the cached sync token. syncToken is now useless
+                let updatedCachedSyncResponse = MXCachedSyncResponse(syncToken: cachedSyncResponse.syncToken,
+                                                                     syncResponse: MXSyncResponse(fromJSON: dictionary as? [AnyHashable : Any]))
+                syncResponseStore.updateSyncResponse(withId: id, syncResponse: updatedCachedSyncResponse)
+                
+                
+            } else {
+                // Use a new chunk
+                NSLog("[MXSyncResponseStoreManager] updateStore: Create a new chunk to store the new sync response. Previous chunk size: \(cachedSyncResponseSize)")
+                let cachedSyncResponse = MXCachedSyncResponse(syncToken: syncToken,
+                                                              syncResponse: newSyncResponse)
+                _ = syncResponseStore.addSyncResponse(syncResponse: cachedSyncResponse)
+            }
+        } else {
+            //  no current sync response, directly save the new one
+            NSLog("[MXSyncResponseStoreManager] updateStore: Start storing sync response")
+            let cachedSyncResponse = MXCachedSyncResponse(syncToken: syncToken,
+                                                         syncResponse: newSyncResponse)
+            _ = syncResponseStore.addSyncResponse(syncResponse: cachedSyncResponse)
+        }
+        
+        // Manage user account data
+        if let accountData = newSyncResponse.accountData {
+            syncResponseStore.accountData = accountData
```

Comment:
Shouldn't we merge account data? Is it right to override that?

Reply:
We have to override it. There is no incremental update of it in matrix.
Overriding is the only way to remove data for example.

Reply:
Can you check again this Manu? Afaics in `-[MXSession handleAccountData:]`, we're updating the existing account data with new events and then store it again.

I'm afraid that some account data events will be ignored here, and even `newSyncResponse.accountData` can be empty dictionary.